packages/docgen/src/get-export-statements.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External dependencies
+ */
+const { SourceFile, SyntaxKind } = require( 'typescript' );
+
+/**
+ * Internal dependencies
+ */
+const { hasExportModifier } = require( './has-modifier' );
+
+/**
+ * ExportAssignment
+ *   - export default 3
+ *   - export default ClassDeclaration;
+ *   - export default fnDeclaration;
+ *
+ * ExportDeclaration:
+ *   - export { a, b } from 'c'
+ *   - export { myDeclaration };
+ *   - export * from './module';
+ *
+ * hasExportModifier
+ *   - export const x = 42
+ *   - export default class {}
+ *   - export default class ClassDeclaration {}
+ *   - export default function() {}
+ *   - export default function myDeclaration() {}
+ *   - export class MyDeclaration {}
+ *   - export function myDeclaration() {}
+ *
```

Comment:
Some cases that don't seem to be covered in this list, are these handled properly?

```js
export { default as foo, createContext as cc } from 'react'
const bar = 'baz'
const myDefault = 'd'
export { bar as quux, myDefault as default }
```

Reply:
I added them here because I want to be clear which condition handles which AST case. I thought your examples could be deduced from `export { a, b } from 'c'` and `export { myDeclaration };`, but it seems that it wasn't. I added them. 

And they're handled properly and tested with unit tests.